### PR TITLE
docs: create GitHub Pages website with Eleventy

### PR DESCRIPTION
**Pull request by jcbhmr**
View preview: https://violet-aileen-6.tiiny.site/
image
image

This PR would...
- [x] create a website
- [x] use eleventy https://www.11ty.dev/
- [x] be basic
- [x] use existing src README content
- [x] add a job when releases are made to deploy the site
- [x] remove the CI to open PRs for the README.md and NOTES.md updates

### CODE REVIEW

File: src/jupyter-datascience-notebooks/README.md
Comment (eitsupi) [inline src/jupyter-datascience-notebooks/README.md:1]: Why change this file?

This file is automatically generated by CI, so manually change is not needed.

**Review by eitsupi — COMMENTED** [review]
Awesome!

But the demo site doesn't seem generated from the latest commit?

File: src/jupyter-datascience-notebooks/README.md
Comment (jcbhmr) [reply src/jupyter-datascience-notebooks/README.md:1]: The CI generates the OPTIONS table and the header and all that. I'm using the README.md here (could be called NOTES.md or index.md or whatever) as the BODY content that goes here:

green part is generated by template, red part is README.md content

image

This way we don't have generated CI artifacts in source control; they get generated at build time for the GitHub Pages site. The "documentationURL" or features would then link to something like: https://violet-aileen-6.tiiny.site/jupyter-datascience-notebooks/ instead of https://github.com/devcontainers-community/templates/tree/main/src/jupyter-datascience-notebooks

**Comment by jcbhmr** [comment]
> But the demo site doesn't seem generated from the latest commit?

🤔 I think the site is showing the right stuff? Maybe I'm misunderstanding what you mean here?

image
image
image

**Comment by jcbhmr** [comment]
I've removed the step to open a PR for the generated NOTES.md ➡ README.md in favor of just using the GitHub Pages website instead. a21218f

File: src/jupyter-datascience-notebooks/README.md
Comment (eitsupi) [reply src/jupyter-datascience-notebooks/README.md:1]: I see.
Perhaps this file could be renamed NOTES.md and a link to the documentation website added to the top-level README of this repository? (This would be a good follow-up PR)

File: .github/workflows/release.yml
Comment (eitsupi) [inline .github/workflows/release.yml:44]: It would be better to test this workflow in PR to see if it works properly.
Could you please add a pull request trigger and change it so that we can check the artifact?
Like this.
https://github.com/rocker-org/website/blob/e59c051b3b2cfbf2171d3817f3ba8ab86514ccc8/.github/workflows/main.yml#L24-L49

**Comment by eitsupi** [comment]
> 🤔 I think the site is showing the right stuff? Maybe I'm misunderstanding what you mean here?

This is my mistake (I thought it was showing the README.md as is). Sorry.

**Comment by jcbhmr** [comment]
Ayyy! 🥳 The preview-docs.yml works!
image

Since it's basically the same as the deploy-docs.yml, I think this is a ✅

**Review by eitsupi — CHANGES_REQUESTED** [review]
> Since it's basically the same as the deploy-docs.yml, I think this is a ✅

Could you use `if` to change the upload location instead of writing almost the same thing in `deploy-docs` and `preview-dows`?
https://github.com/rocker-org/website/blob/e59c051b3b2cfbf2171d3817f3ba8ab86514ccc8/.github/workflows/main.yml#L36-L44

**Comment by jcbhmr** [comment]
You _could_, but you'd also probably want to:
1. use another if statement to only run configure-pages to get a BASE_URL if you're deploying
2. use ANOTHER if statement to only set BASE_URL if you ran configure-pages (or maybe there's some kind of null-coallesce operator?)
3. use two more if statements to branch on if deploying or previewing
4. make them separate jobs so that you only run one with the pages permissions
and by that point i figured it was already _so close_ to two separate workflows that i just split them up 🤷‍♂️

or maybe i got carried away. 🤣

File: .github/workflows/deploy-docs.yml
Comment (eitsupi) [inline .github/workflows/deploy-docs.yml:3]: Another request is to automate website updates. This is currently only performed manually.

**Review by eitsupi — CHANGES_REQUESTED** [review]

**Comment by eitsupi** [comment]
@jcbhmr Why did you merge without my approval?
Am I not needed in this organization?

**Comment by jcbhmr** [comment]
😭 I'm sorry! Yes you are needed to keep me sane and from doing silly impulsive things. Having another person to hold the other one accountable (like you holding me accountable) generally leads to a better outcome.

As to why I merged this... I _niavely assumed_ that all concerns had been met. Obviously, I made another mistake and jumped the gun again. To prevent this from happening again, I've added the requirement of >1 ✅ approvals before merging so that I really _cannot merge_ without a ✅ from you.

---

Was there an additional change that I should make? 🥺

**Comment by jcbhmr** [comment]
I'll also do this on the devcontainers-community/features repo
image

**Comment by eitsupi** [comment]
I apologize that I have been so busy the last few days that I have not had time to look at this. I am hoping to fix this with another PR as I should be able to get some time this evening (it's now morning in Japan).

**Comment by jcbhmr** [comment]
btw it does somewhat work! we have a website now! 🥳
https://devcontainers-community.github.io/templates/

**Comment by eitsupi** [comment]
> Was there an additional change that I should make? 🥺

I think MegaLinter's points are worth noting.

File: .github/workflows/deploy-docs.yml
Comment (eitsupi) [inline .github/workflows/deploy-docs.yml:32]: What is `steps.configure-pages.base_url`?
Maybe `steps.deploy-pages.outputs.page_url`?

File: .github/workflows/deploy-docs.yml
Comment (eitsupi) [inline .github/workflows/deploy-docs.yml:19]: This permission is needed?

**Comment by eitsupi** [comment]
> 1. use another if statement to only run configure-pages to get a BASE_URL if you're deploying. that way you can still `python -m http.server -d zipfolder` and have it WORK (`BASE_URL=/` vs `BASE_URL=/templates/` when generated)

I do not know what this means. Is this really necessary?

I think there was a bug in your workflow (Linter pointed it out but you ignored it) so you didn't actually set up something like `/templates/`.

```log
Run cd docs && npm pack
  cd docs && npm pack
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAGES: true
    BASE_URL: /
```

https://github.com/devcontainers-community/templates/actions/runs/5226625044/jobs/9437466639#step:6:6

**Comment by eitsupi** [comment]
@jcbhmr I have added further branch protection rules.

**Comment by jcbhmr** [comment]
Sounds good! I think I need to change my development habits overall to stop myself from making silly mistakes (like merging too early) for the future. Self improvement! 😜

**Comment by jcbhmr** [comment]
Sounds good! I think I need to change my development habits overall to stop myself from making silly mistakes (like merging too early) for the future. Self improvement! 😜

**Comment by jcbhmr** [comment]
Oh and yeah you're right that I did make the doofus dumdum of misconfiguring GitHub actions and merging it 🤦‍♂️ big time oopsie. Thanks for the constructive criticism ♥️

**Comment by eitsupi** [comment]
Early merging is not a problem in many cases, as it can be corrected later. However, it is definitely worth checking the CI results, I think.

Anyway, updates in #38 seems working fine, so all okay now.

**Pull request merged**
